galley/pkg/config/processor/transforms/ingress/virtualService.go

```diff
@@ -44,18 +45,24 @@ type virtualServiceXform struct {
 	vsByHost  map[string]*syntheticVirtualService
 }
 
-func newVirtualServiceXform(o processing.ProcessorOptions) event.Transformer {
-	xform := &virtualServiceXform{
-		options: o,
+func getVirtualServiceXformProvider() transformer.Provider {
+	inputs := collection.Names{metadata.K8SExtensionsV1Beta1Ingresses}
+	outputs := collection.Names{metadata.IstioNetworkingV1Alpha3Virtualservices}
+
+	createFn := func(o processing.ProcessorOptions) event.Transformer {
+		xform := &virtualServiceXform{
+			options: o,
+		}
+		xform.FnTransform = event.NewFnTransform(
+			inputs,
```

Comment:
Certainly not for this check-in, but I think we can get rid of inputs/outputs from event.Transformer after this (with some additional code changes).